transitions/registration.js

```diff
@@ -181,59 +181,39 @@ module.exports = {
         });
     },
     triggers: {
-        add_patient_id: function(db, doc, cb) {
-            var args = Array.prototype.slice.call(arguments),
-                self = module.exports;
-            cb = args.pop();
-            if (typeof cb !== 'function') {
-                return;
-            }
+        add_patient_id: function(options, cb) {
             // if we already have a patient id then return
-            if (doc.patient_id) {
+            if (options.doc.patient_id) {
                 return;
             }
-            self.setId({db: db, doc: doc}, cb);
+            module.exports.setId(options, cb);
         },
-        add_expected_date: function(db, doc, cb) {
-            var args = Array.prototype.slice.call(arguments),
-                self = module.exports;
-            cb = args.pop();
-            if (typeof cb !== 'function') {
-                return;
-            }
-            self.setExpectedBirthDate(doc);
+        add_expected_date: function(options, cb) {
+            module.exports.setExpectedBirthDate(options.doc);
             cb();
         },
-        add_birth_date: function(db, doc, cb) {
-            var args = Array.prototype.slice.call(arguments),
-                self = module.exports;
-            cb = args.pop();
-            if (typeof cb !== 'function') {
-                return;
-            }
-            self.setBirthDate(doc);
+        add_birth_date: function(options, cb) {
+            module.exports.setBirthDate(options.doc);
             cb();
         },
-        assign_schedule: function(db, doc, cb) {
-            var args = Array.prototype.slice.call(arguments);
-            if (args.length < 4) {
-                cb('Please specify schedule name in settings.');
+        assign_schedule: function(options, cb) {
+            if (!options.params) {
```

Comment:
having trouble tracking down where that `params` comes from, but if it's possible to rename it to `schedules` I'd be happy
Edit: ah, probably from the app_settings. Probably not so changeable then.